testsuite/classlibrary/TestDocument.sc

```diff
@@ -0,0 +1,53 @@
+TestDocument : UnitTest {
+	// autorun doesn't have access to this object's scope
+	// need a public place to communicate the result
+	classvar <>success;
+
+	test_new_document_runs_initAction {
+		var success = false, doc, save;
+		if(Platform.ideName == "scqt") {
+			save = Document.initAction;
+			protect {
+				Document.initAction = { success = true };
+				doc = Document.new;
+				0.1.wait;
+				doc.close;
+			} {
+				Document.initAction = save;
+			};
+			this.assert(success, "Document.new should fire Document.initAction function");
+		} {
+			this.assert(false, "Cannot test Document.new behavior in a non-ScIDE session");
```

Comment:
UnitTest doesn't have a built-in mechanism for marking a test as "skip", but borrowing semantics from other test frameworks, i would expect this to pass, not fail. if we put an "assert false" in here, the test suite will always fail if you are running in an environment without the IDE, which doesn't seem right to me.

i'd suggest replacing this assertion with an empty code block with a comment like `// TODO: skip this test when UnitTest has skip functionality`

Reply:
Ok, fair enough. Should there or should there not be some record of tests that were skipped, and why, in a given run? Eventually yes, I suppose.

Reply:
just remembered there's an open ticket for this: #4384. i propose adding `// TODO: skip` to all skipped tests so we can easily grep for them later when test skipping is implemented.

Reply:
Done.